### PR TITLE
add ktor example

### PR DESCRIPTION
**Pull request by adamsz-lume**
### :pencil: Description
As requested in https://github.com/ExpediaGroup/graphql-kotlin/issues/860#issuecomment-689067996 this PR is adding a working example of ktor implementation with graphql-kotlin

### :link: Related Issues
- subscriptions cannot be simply implemented due to spring connected implementation of com.expediagroup.graphql.spring.execution.ApolloSubscriptionProtocolHandler. What do you think about exporting that class implementation into a separate dependency? It could be then reused in Spark example as well.
- same with the com.expediagroup.graphql.spring.PlaygroundRouteConfiguration. Maybe it's worth take it out as a simple class/function which returns the html which than can be used in other frameworks.
- another idea would be to create a core library for those

I hope this PR will enable a nice ktor feature implementation of graphql.

### CODE REVIEW

**Comment by smyrick** [comment]
@adamsz-lume Thank you for your contributions! To make it more clear this is for server code, let's rename the module to `examples/ktor-server`. I know the current `spring` module doesn't follow this pattern but I think we will rename this to make it more clear

File: examples/ktor/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+val ktor_version: String by project
+val logback_version: String by project
+
+description = "An Example GraphQL service served by a Spark HTTP server"
```
Comment (smyrick) [inline examples/ktor/build.gradle.kts:4]:
```suggestion
description = "An Example GraphQL service served by a Ktor HTTP server"
```

File: examples/ktor/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+val kotlinCoroutinesVersion: String by project
+dependencies {
+    implementation("io.ktor:ktor-server-core:$ktor_version")
+    implementation("io.ktor:ktor-server-netty:$ktor_version")
+    implementation("ch.qos.logback:logback-classic:$logback_version")
+
+    implementation("com.expediagroup", "graphql-kotlin-schema-generator")
+    implementation("org.jetbrains.kotlinx", "kotlinx-coroutines-jdk8", kotlinCoroutinesVersion)
+    testImplementation("org.slf4j", "slf4j-log4j12", "1.7.30")
```
Comment (smyrick) [inline examples/ktor/build.gradle.kts:22]: We can also add this version to the `gradle.properties` if we have the file

File: examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/GraphQLModule.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import io.ktor.routing.get
+import io.ktor.routing.post
+import io.ktor.routing.routing
+import java.lang.IllegalStateException
+
+fun Application.graphQLModule() {
+    install(Routing)
+
+    routing {
+        get("hello") {
```
Comment (smyrick) [inline examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/GraphQLModule.kt:19]: We can remove this route

File: examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/HelloQueryService.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.graphql.examples.ktor.schema
+
+class HelloQueryService {
```
Comment (smyrick) [inline examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/HelloQueryService.kt:18]: If the route is deleted we can delete the service too

**Review by smyrick — CHANGES_REQUESTED** [review]

**Comment by smyrick** [comment]
Let's also edit the `examples/README.md` file to list the new module

File: examples/ktor/build.gradle.kts (hunk 2 above)
Comment (adamsz-lume) [reply examples/ktor/build.gradle.kts]: just spotted it was not needed as logback is used anyway

File: examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/GraphQLModule.kt (hunk 3 above)
Comment (adamsz-lume) [reply examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/GraphQLModule.kt]: it's gone

File: examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/HelloQueryService.kt (hunk 4 above)
Comment (adamsz-lume) [reply examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/HelloQueryService.kt]: This service is part of the graphql schema per spark example. It is convenient to have it to make a quick query.

Comment (smyrick) [reply examples/ktor/src/main/kotlin/com/expediagroup/graphql/examples/ktor/schema/HelloQueryService.kt]: Ah I see, I thought this was for the hello http route

**Review by smyrick — APPROVED** [review]

**Comment by dariuszkuc** [comment]
Very nice!

**Pull request merged**